File: 0x13-javascript_objects_scopes_closures/1-rectangle.js
#!/usr/bin/node

class Rectangle {
  constructor (w, h) {
    this.width = w;
    this.height = h;
  }
}

const rect1 = new Rectangle(4, 5);
console.log(rect1);
console.log(rect1.width);
console.log(rect1.height);

const rect2 = new Rectangle(4, -6);
console.log(rect1);
console.log(rect2.width);
console.log(rect2.height);

File: 0x08-python-more_classes/1-rectangle.py
#!/usr/bin/python3

"""Define a rectangle"""


class Rectangle:
    """representation of rectangle"""

    def __init__(self, width=0, height=0):
        """initializing the rectangle by
        setting object of width and height"""

        self.width = width
        self.height = height

        @property
        def width(self):
            """getter for private instance for attribute width"""
            return self.__width

        @width.setter
        def width(self, value):
            """setter for private instance attribute for width"""
            if type(value) is not int:
                raise TypeError("width must be an integer")
            if value < 0:
                raise ValueError("width must be >= 0")
                self.__width = value

        @property
        def height(self):
            """getter for private instance attribute for height"""
            return self.__height

        @height.setter
        def height(self, value):
            """setter for private instance attribute for height"""
            if type(value) is not int:
                raise TypeError("height must be an integer")
            if value < 0:
                raise ValueError("height must be >= 0")
                self.__height = value
